### PR TITLE
[CBRD-23790] [TDE] Support multi-national languages

### PR DESCRIPTION
**Pull request by wrlawodms**
http://jira.cubrid.org/browse/CBRD-23790

In the TDE project, many messages are added including usage and error messages. However, now only Korean and English are supported. It must be addressed.

The addressed encodings here:
- de_DE.utf8
- es_ES.utf8
- fr_FR.utf8
- it_IT.utf8
- ja_JP.utf8
- km_KH.utf8
- ko_KR.euckr
- ro_RO.utf8
- tr_TR.utf8
- vi_VN.utf8
- zh_CN.utf8

### CODE REVIEW

File: msg/de_DE.utf8/utils.msg [hunk 1]
```diff
@@ -1191,3 +1203,23 @@ gültigen Optionen:\n\
       --resume                  Wiederaufnahme der Berechnung Prüfsumme\n\
       --schema-only             überprüfen nur Schemakonsistenz\n\
       --cont-on-error           weiter auf Berechnungsfehler\n
+
+$set 57 MSGCAT_UTIL_SET_TDE
+21 Geben Sie das DBA-Passwort ein:
+25 Die in der Datenbank festgelegten Schlüsselinformationen konnten nicht abgerufen werden\n
+26 Versuch, den Schlüssel vom Schlüssel (Index: %1$d) zum Schlüssel (Index: %2$d) zu ändern.\n
```
Comment (wrlawodms) [inline msg/de_DE.utf8/utils.msg:1210]:
```suggestion
26 Versuch, den Schlüssel vom Schlüssel (Index: %1$d) zum Schlüssel (Index: %2$d) zu ändern ..\n
```

File: msg/es_ES.utf8/utils.msg [hunk 2]
```diff
@@ -453,7 +464,7 @@ $set 12 MSGCAT_UTIL_SET_LOADDB
 117 Carga fallida.\n
 118 La longitud máxima del nombre de la clase es %1$d bytes.\n
 119 Total de %1$8d objeto(s) verificado, %2$d objeto(s) falló. \n
-120 del archivo de datos no contiene '%%class' especificador o '--table' falta la opción \n
+120 del archivo de datos no contiene '%% class' especificador o '--table' falta la opción \n
```
Comment (JoohoK) [inline msg/es_ES.utf8/utils.msg:467]: if you intended this change, I think this change will be applied to other msg files.

Comment (wrlawodms) [reply msg/es_ES.utf8/utils.msg:467]: No, I didn't. Thanks

File: msg/ja_JP.utf8/utils.msg [hunk 3]
```diff
@@ -1193,3 +1205,23 @@ checksumdb: レプリケーションの整合性を確認します。\n\
       --resume                  チェックサムの計算を再開\n\
       --schema-only             スキーマ整合性のみをチェックします。\n\
       --cont-on-error           計算エラーが発生したときに継続\n
+
+$set 57 MSGCAT_UTIL_SET_TDE
+21 DBAパスワードを入力します。
+25 データベースに設定されているキー情報を取得できませんでした\n
+26 キーをキー（インデックス：%1$d）からキー（インデックス：%2$d）に変更しようとしています。\n
```
Comment (wrlawodms) [inline msg/ja_JP.utf8/utils.msg:1212]:
```suggestion
26 キーをキー（インデックス：%1$d）からキー（インデックス：%2$d）に変更しようとしています。。\n
```

File: msg/ja_JP.utf8/utils.msg [hunk 4]
```diff
@@ -279,7 +288,8 @@ backupdb: データーベースをバックアップします。\n\
       --no-check               データーベースの整合性を確認する作業を実行しません。;　デフォルト:実行\n\
   -t, --thread-count=COUNT     バックアップをするスレッド数。;　デフォルト:自動\n\
   -z, --compress               バックアップボリュームを圧縮します。;　デフォルト:圧縮しない\n\
-      --sleep-msecs            １メガバイトを読むたびにNミリセコンドを休みます。;　デフォルト: 0 \n
+      --sleep-msecs            １メガバイトを読むたびにNミリセコンドを休みます。;　デフォルト: 0 \n\
+  -k、--separate_keys          TDEのキーファイル（_keys）はバックアップボリュームに含まれていませんが、代わりにファイルに分割されています\n
```
Comment (wrlawodms) [inline msg/ja_JP.utf8/utils.msg:292]:
```suggestion
  -k, --separate_keys           TDEのキーファイル（_keys）はバックアップボリュームに含まれていませんが、代わりにファイルに分割されています\n
```

File: msg/ja_JP.utf8/utils.msg [hunk 5]
```diff
@@ -1140,7 +1151,8 @@ restoreslave: スレーブデータベースを復元します。\n\
       --list                        復元せずにバックアップボリュームのリストを表示\n\
   -B, --backup-file-path=PATH       PATHには、バックアップボリュームのディレクトリを復元することを意味します\n\
   -o, --output-file=FILE            FILEに出力メッセージをリダイレクトします; デフォルト: none\n\
-  -u, --use-database-location-path  データベース・ロケーション・ファイルで指定したパスにデータベースとログボリュームを復元します\n
+  -u, --use-database-location-path  データベース・ロケーション・ファイルで指定したパスにデータベースとログボリュームを復元します\n\
+  -k、--keys-file-path              復元中に使用されるTDEのキーファイル（_keys）のパス\n
```
Comment (wrlawodms) [inline msg/ja_JP.utf8/utils.msg:1155]:
```suggestion
  -k、--keys-file-path               復元中に使用されるTDEのキーファイル（_keys）のパス\n
```

**Review by JoohoK — APPROVED** [review]

File: msg/ja_JP.utf8/utils.msg [hunk 6]
```diff
@@ -397,7 +407,8 @@ restoredb: データベース復旧\n\
   -l, --level=LEVEL                 復旧に使うバックアップのレベル; デフォルト: 0(Full backup)\n\
   -p, --partial-recovery            アーカイブログがなければ強制にパーシャルリカバリーを実行する。\n\
   -o, --output-file=FILE            出力メッセージがセーブされるファイル; デフォルト: なし\n\
-  -u, --use-database-location-path  データベース位置ファイルに設定されたパスにデータベースボリュームとログボリュームを復旧\n
+  -u, --use-database-location-path  データベース位置ファイルに設定されたパスにデータベースボリュームとログボリュームを復旧\n\
+  -k、--keys-file-path               復元中に使用されるTDEのキーファイル（_keys）のパス\n
```
Comment (hornetmj) [inline msg/ja_JP.utf8/utils.msg:411]: indentation. plz check using vi.

File: msg/km_KH.utf8/utils.msg [hunk 7]
```diff
@@ -1193,3 +1205,23 @@ valid options:\n\
...
+27 The key has been changed from the key (index: %1$d) to the key (index: %2$d)\n
+28 The key set on the database cannot be deleted\n
+29 The key (index: %1$d) has been deleted\n
+30 A new key has been generated - key index: %1$d created on %2$s
+60 \
+tde: Transparent Data Encryption (TDE) management\n\
+usage: %1$s tde OPERATION(-s|-n|-d|-c) [OPTION] database-name\n\
+\n\
+valid options:\n\
+    -p, --dba-password=PASS      password of the DBA user; will prompt if don't specify\n\
```
Comment (hornetmj) [inline msg/km_KH.utf8/utils.msg:1222]: need a space before 'password'

File: msg/ko_KR.euckr/utils.msg [hunk 8]
```diff
@@ -1180,3 +1192,23 @@ valid options:\n\
...
+27 Ű (�ε���: %1$d)���� Ű (�ε���: %2$d)�� ����Ǿ��ϴ�.\n
+28 �����ͺ��̽��� ��ϵ� Ű�� ����� �� ���ϴ�.\n
+29 �ε���: %1$d Ű�� ����Ͽ���ϴ�.\n
+30 ��ο� Ű�� ��Ǿ��ϴ�. Ű �ε���: %1$d, �� �ð�: %2$s
+60 \
+tde: Transparent Data Encryption (TDE) ��\n\
+usage: %1$s tde <operation> [�ɼ�] <�����ͺ��̽�-�̸�>\n\
+\n\
+valid options:\n\
+    -p, --dba-password=PASS         DBA �н����; ������� ��� ��� �����Ʈ\n\
```
Comment (hornetmj) [inline msg/ko_KR.euckr/utils.msg:1209]: indent check before 'DBA'

File: msg/ro_RO.utf8/cubrid.msg [hunk 9]
```diff
@@ -1314,8 +1314,25 @@ $ LOADDB
 1245 Încărcarea atributelor de tip "class" nu este suportată in modul client-server.
 1246 Nu se poate obține adresa și numele canonic pentru ("%1$s") în /etc/hosts. Vă rugăm să verificați permisiunea fișierului sau dacă există numele gazdei.
 1247 Momentul specificat (%1$s) trebuie să îl succeadă pe cel al copiei de rezervă specificate(%2$s).
+1248 Fișier cheie nevalid: %1$s
+1249 Nu se poate găsi cheia (index: %1$d).
+1250 Cheia master setată în baza de date și ceea ce este dat nu se potrivesc. Indicele cheie setat în baza de date: %1$d
+1251 Eroare la criptarea TDE.
+1252 Eroare la decriptarea TDE.
+Modulul TDE 1253 nu este încărcat.
```
Comment (hornetmj) [inline msg/ro_RO.utf8/cubrid.msg:1322]: start with 1253.

File: msg/ro_RO.utf8/utils.msg [hunk 10]
```diff
@@ -1219,3 +1231,23 @@ opțiuni valide:\n\
...
+27 Cheia a fost schimbată de la cheie (index: %1$d) la cheie (index: %2$d)\n
+28 Cheia setată în baza de date nu poate fi ștearsă\n
+29 Cheia (index: %1$d) a fost ștearsă\n
+30 A fost generată o nouă cheie - index cheie: %1$d creat pe%2$s
+60 \
+tde: gestionarea criptării transparente a datelor (TDE)\n\
+utilizare: %1$s tde OPERARE (-s|-n|-d|-c) [OPȚIUNE] nume bază de date\n\
+\n\
+opțiuni valide:\n\
+    -p, --dba-password=PASS    parola utilizatorului DBA; va solicita dacă nu specificați\n\
```
Comment (hornetmj) [inline msg/ro_RO.utf8/utils.msg:1248]: indent check before parola

File: msg/tr_TR.utf8/cubrid.msg [hunk 11]
```diff
@@ -1314,8 +1314,25 @@ $ LOADDB
...
+1250 Veritabanında ayarlanan ana anahtar ve verilenler eşleşmiyor. Veritabanında ayarlanan anahtar dizini: %1$d
+1251 TDE şifreleme sırasında hata.
+1252 TDE şifresini çözerken hata oluştu.
+1253 TDE Modülü yüklenmemiş.
+1254 Anahtar oluşturulamıyor.
+1255 TDE modülü yüklenemiyor. TDE özelliğini kullanamazsınız.\n\
+Anahtar dosyasının (_keys) yolunu kontrol edin ve uygun anahtarı içerdiğinden emin olun.
+1256 Anahtar dosyası (_keys) kopyalanamıyor. Kopyalanan veri tabanında TDE özelliğini kullanamazsınız.
+1257 Anahtar dosyası (_keys) yedeklenemiyor. Bu yedekleme birimine bağlı olarak geri yüklenen veritabanında TDE özelliğini kullanamazsınız.
+1258%1$s veritabanında anahtar kümesi bulunamıyor. Yedekleme birimindeki anahtar dosya kullanılacaktır.
```
Comment (hornetmj) [inline msg/tr_TR.utf8/cubrid.msg:1328]: need a space '1258%1$s'

File: msg/tr_TR.utf8/utils.msg [hunk 12]
```diff
@@ -1186,3 +1198,23 @@ geçerli seçenekler:\n\
...
       --cont-on-error           hesaplama hatası devam\n
+
+$set 57 MSGCAT_UTIL_SET_TDE
+21 DBA şifresini girin:
+25 Veritabanındaki anahtar bilgi kümesini alamadı\n
+26 Anahtarı anahtardan (dizin: %1$d) anahtara (dizin: %2$d) değiştirmeye çalışılıyor ..\n
+27 Anahtar, anahtardan (dizin: %1$d) anahtara (dizin: %2$d) değiştirildi\n
+28 Veritabanındaki anahtar seti silinemez\n
+29 Anahtar (dizin: %1$d) silindi\n
+30 Yeni bir anahtar oluşturuldu - anahtar dizini: %1$d%2$s üzerinde oluşturuldu
```
Comment (hornetmj) [inline msg/tr_TR.utf8/utils.msg:1209]: is '%1$d%2$s' ok?

File: msg/vi_VN.utf8/utils.msg [hunk 13]
```diff
@@ -1193,3 +1205,23 @@ valid options:\n\
...
+27 The key has been changed from the key (index: %1$d) to the key (index: %2$d)\n
+28 The key set on the database cannot be deleted\n
+29 The key (index: %1$d) has been deleted\n
+30 A new key has been generated - key index: %1$d created on %2$s
+60 \
+tde: Transparent Data Encryption (TDE) management\n\
+usage: %1$s tde OPERATION(-s|-n|-d|-c) [OPTION] database-name\n\
+\n\
+valid options:\n\
+    -p, --dba-password=PASS      password of the DBA user; will prompt if don't specify\n\
```
Comment (wrlawodms) [inline msg/vi_VN.utf8/utils.msg:1222]:
```suggestion
    -p, --dba-password=PASS       password of the DBA user; will prompt if don't specify\n\
```

**Review by hornetmj — APPROVED** [review]

File: msg/de_DE.utf8/cubrid.msg [hunk 14]
```diff
@@ -1314,8 +1314,25 @@ $ LOADDB
 1245 Das Laden von Aktualisierungen von Klassenattributen aus Objektdateien wird im CS-Modus nicht unterstützt.
 1246 Adresse und kanonischer Name für ("%1$s") in /etc/hosts können nicht abgerufen werden. Bitte überprüfen Sie die Berechtigung der Datei oder ob der Hostname vorhanden ist.
 1247 Die angegebene Zeit (%1$s) muss später als (%2$s), die Zeit der angegebenen Sicherung.
+1248 Ungültige Schlüsseldatei : %1$s
+1249 Der Schlüssel kann nicht gefunden werden (Index: %1$d).
```
[the remark this answers is not in the document]
Comment (wrlawodms) [reply msg/de_DE.utf8/cubrid.msg:1318]:
```suggestion
1249 Der Schlüssel kann nicht gefunden werden (index: %1$d).
```

File: msg/de_DE.utf8/utils.msg [hunk 15]
```diff
@@ -261,6 +264,12 @@ Vorhandene Befehle:\n\
 $set 2 MSGCAT_UTIL_SET_BACKUPDB
 30 WARNUNG: -t Option wird in Stand-Alone-Ausführung ignoriert.\n
 31 FEHLER: Destination-Pfad existiert nicht oder ist kein Verzeichnis.\n
+32 Mit der Option --separate-keys wird die Schlüsseldatei (_keys) vom Sicherungsdatenträger getrennt.\n\
+Wenn die Schlüsseldatei verloren geht, funktioniert die Wiederherstellung (restoreb) möglicherweise nicht ordnungsgemäß, selbst wenn das Sicherungsvolume vorhanden ist.\n
```
[the remark this answers is not in the document]
Comment (wrlawodms) [reply msg/de_DE.utf8/utils.msg:268]:
```suggestion
Wenn die Schlüsseldatei verloren geht, funktioniert die Wiederherstellung (restoredb) möglicherweise nicht ordnungsgemäß, selbst wenn das Sicherungsvolume vorhanden ist.\n
```

File: msg/de_DE.utf8/utils.msg (hunk 1 above)
Comment (wrlawodms) [inline msg/de_DE.utf8/utils.msg]:
```suggestion
26 Versuch, den Schlüssel vom Schlüssel (Index: %1$d) zum Schlüssel (Index: %2$d) zu ändern..\n
```

File: msg/de_DE.utf8/utils.msg [hunk 16]
```diff
@@ -1191,3 +1203,23 @@ gültigen Optionen:\n\
       --resume                  Wiederaufnahme der Berechnung Prüfsumme\n\
       --schema-only             überprüfen nur Schemakonsistenz\n\
       --cont-on-error           weiter auf Berechnungsfehler\n
+
+$set 57 MSGCAT_UTIL_SET_TDE
+21 Geben Sie das DBA-Passwort ein:
+25 Die in der Datenbank festgelegten Schlüsselinformationen konnten nicht abgerufen werden\n
+26 Versuch, den Schlüssel vom Schlüssel (Index: %1$d) zum Schlüssel (Index: %2$d) zu ändern.\n
+27 Der Schlüssel wurde vom Schlüssel (Index: %1$d) zum Schlüssel (Index: %2$d) geändert\n
```
[the remark this answers is not in the document]
Comment (wrlawodms) [reply msg/de_DE.utf8/utils.msg:1211]:
```suggestion
27 Der Schlüssel wurde vom Schlüssel (index: %1$d) zum Schlüssel (index: %2$d) geändert\n
```

File: msg/de_DE.utf8/utils.msg [hunk 17]
```diff
@@ -1191,3 +1203,23 @@ gültigen Optionen:\n\
...
+28 Der in der Datenbank festgelegte Schlüssel kann nicht gelöscht werden\n
+29 Der Schlüssel (Index: %1$d) wurde gelöscht\n
```
[the remark this answers is not in the document]
Comment (wrlawodms) [reply msg/de_DE.utf8/utils.msg:1213]:
```suggestion
29 Der Schlüssel (index: %1$d) wurde gelöscht\n
```

File: msg/es_ES.utf8/utils.msg [hunk 18]
```diff
@@ -261,6 +264,12 @@ Comando disponible:\n\
 $set 2 MSGCAT_UTIL_SET_BACKUPDB
 30 ADVERTENCIA: -t opcion es ignorada en modo de ejecucion independiente.\n
 31 ERROR: Calle de destino no existe o no es un directorio.\n
+32 Al usar la opción --separate-keys, separa el archivo de claves (_keys) del volumen de respaldo.\n\
+Si se pierde el archivo de claves, es posible que la restauración (restauradab) no funcione correctamente incluso si existe el volumen de copia de seguridad.\n
```
[the remark this answers is not in the document]
Comment (wrlawodms) [reply msg/es_ES.utf8/utils.msg:268]:
```suggestion
Si se pierde el archivo de claves, es posible que la restauración (restoredb) no funcione correctamente incluso si existe el volumen de copia de seguridad.\n
```

File: msg/es_ES.utf8/utils.msg (hunk 2 above)
Comment (wrlawodms) [inline msg/es_ES.utf8/utils.msg]:
```suggestion
120 del archivo de datos no contiene '%%class' especificador o '--table' falta la opción \n
```

File: msg/ro_RO.utf8/cubrid.msg (hunk 9 above)
Comment (wrlawodms) [reply msg/ro_RO.utf8/cubrid.msg:1322]:
```suggestion
1253 Modulul TDE nu este încărcat.
```

File: msg/tr_TR.utf8/cubrid.msg (hunk 11 above)
Comment (wrlawodms) [reply msg/tr_TR.utf8/cubrid.msg:1328]:
```suggestion
1258 %1$s veritabanında anahtar kümesi bulunamıyor. Yedekleme birimindeki anahtar dosya kullanılacaktır.
```

File: msg/tr_TR.utf8/utils.msg (hunk 12 above)
Comment (wrlawodms) [reply msg/tr_TR.utf8/utils.msg:1209]:
```suggestion
30 Yeni bir anahtar oluşturuldu - anahtar dizini: %1$d %2$s üzerinde oluşturuldu
```

**Pull request merged**